src/main/java/seedu/address/logic/commands/AddTuitionTaskCommand.java

```diff
@@ -19,67 +20,75 @@
-    private final TuitionTask toAdd;
     private final Index targetIndex;
+    private final LocalDateTime taskdateTime;
+    private final String duration;
+    private final String description;
 
+    private TuitionTask toAdd;
     private TuitionSchedule tuitionSchedule;
-    private Tutee associatedPerson;
+    private Tutee associatedTutee;
 
     /**
-     * Creates an AddTuition to add the specified {@code Task} which is associated to {@code Person}.
+     * Creates an AddTuition to add the specified {@code Task} which is associated to {@code Tutee}.
      */
-    public AddTuitionTaskCommand(TuitionTask task, Index indexOfAssociatedPerson) {
-        requireNonNull(task);
-        toAdd = task;
-        targetIndex = indexOfAssociatedPerson;
+    public AddTuitionTaskCommand(Index targetIndex, LocalDateTime taskDateTime, String duration, String description) {
+        requireNonNull(taskDateTime);
+        requireNonNull(duration);
+        requireNonNull(description);
+        this.taskdateTime = taskDateTime;
+        this.duration = duration;
+        this.description = description;
+        this.targetIndex = targetIndex;
     }
 
     @Override
     public CommandResult executeUndoableCommand() {
-        requireNonNull(tuitionSchedule);
         tuitionSchedule.addTask(toAdd);
         model.addTask(toAdd);
         return new CommandResult(String.format(MESSAGE_SUCCESS, toAdd));
     }
 
     @Override
     protected void preprocessUndoableCommand() throws CommandException {
-        findAssociatedPerson();
-        assert(associatedPerson.getTuitionSchedule() != null);
-        tuitionSchedule = associatedPerson.getTuitionSchedule();
+        associatedTutee = getAssociatedTutee();
+        requireNonNull(associatedTutee.getTuitionSchedule());
+        tuitionSchedule = associatedTutee.getTuitionSchedule();
+        toAdd = new TuitionTask(associatedTutee, taskdateTime, duration, description);
     }
 
     /**
-     * Locates {@code Person} that will be associated with the tuition task.
-     *
-     * @throw commandException if invalid person index was given.
+     * Returns the {@code Tutee} object that is pointed by the index as shown in the last displayed conatct list.
      */
-    private void findAssociatedPerson() throws CommandException {
+    private Tutee getAssociatedTutee() throws CommandException {
         List<Person> lastShownList = model.getFilteredPersonList();
 
         if (targetIndex.getZeroBased() >= lastShownList.size()) {
             throw new CommandException(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
         }
-
-        associatedPerson = (Tutee) lastShownList.get(targetIndex.getZeroBased());
+        Person associatedPerson = lastShownList.get(targetIndex.getZeroBased());
+        if (!(associatedPerson instanceof Tutee)) {
+            throw new CommandException(Messages.MESSAGE_INVALID_TUTEE_INDEX);
+        }
+        return (Tutee) lastShownList.get(targetIndex.getZeroBased());
     }
```

Comment:
finding tutee given index. Throws an exception if the given index does not point to a Tutee object. (just a normal person)